pkg/northbound/gnmi/subscribe.go

```diff
@@ -126,11 +127,11 @@ func listenOnChannel(stream gnmi.GNMI_SubscribeServer, mgr *manager.Manager, has
 		} else {
 			subs := subscribe.Subscription
 			//FAST way to identify if target and subscription is present
-			subsStr := make(map[string]struct{})
+			subsStr := make([]*regexp.Regexp, 0)
```

Comment:
I'd suggest we keep it to a map, looking on the web it seems the the map with the struct{} is faster to index against then the list for linear search. that's why I sudden it in the first place. also keeps code complexity lower

Reply:
Map gives no advantage as we don't ever lookup by RegExp. We have to test each path to see if it rexexp matches any one of the subs.

Refactored the iteration over regexps out in to its own function